src/libraries/Microsoft.Extensions.Logging/src/CompatibilitySuppressions.xml

```diff
@@ -0,0 +1,402 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```

Comment:
Why is this file being added? We didn't need it for any other library. What is different/special about this one?

Reply:
No idea. CI just started randomly failing after nullable annotations were added

Reply:
@joperezr and I spent time debugging this today. API compat is complaining because we are breaking the API here.  Here's an example:

```diff
- public LoggerFilterRule(string providerName, string categoryName, Microsoft.Extensions.Logging.LogLevel? logLevel, System.Func<string, string, Microsoft.Extensions.Logging.LogLevel, bool> filter) { }
+ public LoggerFilterRule(string? providerName, string? categoryName, Microsoft.Extensions.Logging.LogLevel? logLevel, System.Func<string?, string?, Microsoft.Extensions.Logging.LogLevel?, bool>? filter) { }
```

Putting `?` on reference types is fine in these nullable annotations. But there are APIs where we are putting `Microsoft.Extensions.Logging.LogLevel?` where it wasn't nullable before. This is an API breaking change because LogLevel is an enum - i.e. a value type. We can't be changing the nullability of value types in these changes.

So please revert the CompatibilitySuppressions.xml file and all the changing of LogLevel nullability.

Reply:
Riiight, missed that, thanks! 
But I think, it's annotated like that in `Logging.Abstractions`, and `LogLevel?` is valid there. Should there be another PR for that, or it's better to leave `LogLevel` in `Loggins` as is?

Reply:
> But I think, it's annotated like that in Logging.Abstractions, and LogLevel? is valid there.

Can you point to where?

> Should there be another PR for that, or it's better to leave LogLevel in Loggins as is?

It is better to leave LogLevel exactly as it is defined today. We can't be changing the nullability of an enum without a breaking change.

Reply:
> Can you point to where?

Seems like I was mistaken. `LoggerFilterRule.LogLevel` is defined as nullable, so I thought that it also should be nullable in the `Filter`.

Reverted the changes